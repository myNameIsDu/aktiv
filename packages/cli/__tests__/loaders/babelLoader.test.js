const babelLoader = require('../../ak-webpack-config/lib/loaders/babelLoader');
const { localBuildEnv, browserTarget } = require('../../config/index');
const getPresets = require('../../ak-webpack-config/presets/index.js');

const path = require('path');

describe('loader-babelLoader', () => {
    const workDir = './';

    it('默认babel配置', () => {
        const babelLoaderOptions = babelLoader({ workDir }, {});

        expect(babelLoaderOptions).toMatchSnapshot();
    });

    it('传入babelInclude应当包含在include配置内', () => {
        const babelInclude = ['./a'];
        const babelLoaderConfig = babelLoader(
            {
                babelInclude,
                workDir,
            },
            {},
        );

        // eslint-disable-next-line no-plusplus
        for (let index = 0; index < babelLoaderConfig.length; index++) {
            const element = babelLoaderConfig[index];

            expect(element.include).toEqual(
                expect.arrayContaining([...babelInclude.map(v => path.resolve(workDir, v))]),
            );
        }
    });

    it('local环境并且开启liveReload则添加react-refresh', () => {
        const localPreset = getPresets(localBuildEnv, browserTarget);

        const babelLoaderConfig = babelLoader(
            {
                workDir,
                liveReload: true,
            },
            localPreset,
        );

        // eslint-disable-next-line no-plusplus
        for (let index = 0; index < babelLoaderConfig.length; index++) {
            const element = babelLoaderConfig[index];

            // @ts-ignore 测试options
            expect(element.use[0].options.plugins).toEqual(
                expect.arrayContaining([require.resolve('react-refresh/babel')]),
            );
        }
    });
    it('开启antdModuleLazyOff应该添加import插件', () => {
        const babelLoaderConfig = babelLoader(
            {
                workDir,
                antdModuleLazyOff: true,
            },
            {},
        );

        // eslint-disable-next-line no-plusplus
        for (let index = 0; index < babelLoaderConfig.length; index++) {
            const element = babelLoaderConfig[index];

            // @ts-ignore 测试options
            expect(element.use[0].options.plugins).toEqual(
                expect.arrayContaining([
                    [
                        require.resolve('babel-plugin-import'),
                        {
                            libraryName: 'antd',
                            libraryDirectory: 'es',
                            style: true,
                        },
                    ],
                ]),
            );
        }
    });
});
