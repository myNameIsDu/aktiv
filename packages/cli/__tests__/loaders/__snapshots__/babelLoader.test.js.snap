// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader-babelLoader 默认babel配置 1`] = `
Array [
  Object {
    "include": Array [
      "/Users/du/Documents/my/my-cli/du-formula/src",
    ],
    "test": /\\\\\\.\\(js\\|jsx\\)\\$/,
    "use": Array [
      Object {
        "loader": "babel-loader",
        "options": Object {
          "cacheCompression": false,
          "cacheDirectory": true,
          "cwd": "/Users/du/Documents/my/my-cli/du-formula/packages/cli",
          "plugins": Array [
            "@babel/plugin-transform-runtime",
            Array [
              "@babel/plugin-proposal-decorators",
              Object {
                "legacy": true,
              },
            ],
          ],
          "presets": Array [
            Array [
              "@babel/preset-env",
              Object {
                "bugfixes": true,
                "corejs": Object {
                  "proposals": true,
                  "version": 3.18,
                },
                "useBuiltIns": "usage",
              },
            ],
            Array [
              "@babel/preset-react",
              Object {
                "runtime": "automatic",
              },
            ],
          ],
        },
      },
    ],
  },
  Object {
    "include": Array [
      "/Users/du/Documents/my/my-cli/du-formula/src",
    ],
    "test": /\\\\\\.\\(ts\\|tsx\\)\\$/,
    "use": Array [
      Object {
        "loader": "babel-loader",
        "options": Object {
          "cacheCompression": false,
          "cacheDirectory": true,
          "cwd": "/Users/du/Documents/my/my-cli/du-formula/packages/cli",
          "plugins": Array [
            "@babel/plugin-transform-runtime",
            Array [
              "@babel/plugin-proposal-decorators",
              Object {
                "legacy": true,
              },
            ],
          ],
          "presets": Array [
            Array [
              "@babel/preset-env",
              Object {
                "bugfixes": true,
                "corejs": Object {
                  "proposals": true,
                  "version": 3.18,
                },
                "useBuiltIns": "usage",
              },
            ],
            Array [
              "@babel/preset-react",
              Object {
                "runtime": "automatic",
              },
            ],
            "@babel/preset-typescript",
          ],
        },
      },
    ],
  },
]
`;
